saplings/circuits/src/components/circuitDetails/CircuitDetails.js

```diff
@@ -73,7 +73,9 @@ const CircuitDetails = () => {
           );
           setNodes(filteredNodes);
         } catch (e) {
-          throw Error(`Unable to fetch nodes from the node registry: ${e}`);
+          throw Error(
```

Comment:
Could any of the other functions in the try block throw an error? It might be safer to return the `e.json.message` from the API call functions